### ISSUE TITLE
Classifier-free Diffusion Guidance

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#160]
## TL; DR

- With implicit classifier formulation, we can achieve similar effects (fidelity-diversity trade-off) as classifier-guided diffusion *without* training a separate classifier (using noisy images).

## Motivation

- Classifier-guided diffusion models entail several blind spots: inconvenience of training a separate classifier, connection to gradient-based adversarial attack on classifier and resemblance to GANs (which has already been shown to perform extremely well)

## Prior Works

- **DDPM** (connect diffusion models to denoising score matching and Langevin sampling)
- **Improved DDPM** (technical modifications such as cosine scheduling and learned variance to improve likelihood estimation and accelerate sampling)
- **ADM** (architecture ablation and classifier guidance for higher sample quality)

## Method

- Reformulate classifier guidance using Bayes rule:
![image](https://user-images.githubusercontent.com/46308407/153552298-83a97a68-1901-4eb4-affe-9ee9acba207e.png)

- Hence, we can mimic classifier guidance using two generative models: conditional and unconditional diffusion models.
- In practice, single neural network can represent both models with condition set to *zero* when employing the unconditional version.
- We increase the likelihood for class-conditional sample while ***reducing*** the likelihood for unconditional samples (hence draw samples from the part of manifold where higher probability density is assigned for corresponding class)
- This is similar to low-temperature sampling or truncation trick.
![image](https://user-images.githubusercontent.com/46308407/153552573-50d0a144-af80-4e53-b623-bb4f27625e61.png)

## Experiments

- Raising the weight gives better fidelity (hence higher IS) but hurts diversity (hence FID)
- Sweet spot for FID is w=0.1, while IS is linearly improved.

![image](https://user-images.githubusercontent.com/46308407/153552650-12b4e929-dcdf-41f1-9ce3-6d6adbcfcde8.png)
![image](https://user-images.githubusercontent.com/46308407/153552670-6374520e-87b3-4e3a-bb66-f26d870e7fba.png)
![image](https://user-images.githubusercontent.com/46308407/153552694-0b4bd343-e1c8-44f7-933a-a4b93fd16507.png)
![image](https://user-images.githubusercontent.com/46308407/153552717-2e9f218e-c75f-4b19-93fb-bd14663fe756.png)

## Comments

- Excellent formulation of *implicit classifier*.
- The idea is so simple yet very effective. Impressive result on *near linear* trend for FID/IS trade-off. The authors must have been pleased to see such a neat plot.
- (Theoretical) Probabilistic interpretability, in my personal opinion, is one of the coolest part of diffusion model family. Reducing likelihood for unconditional samples while raising that for conditional samples for corresponding magnitude sounds reasonable.
